### PR TITLE
Add constraint for minimum AU Medicare Card Number length of 10 characters (FHIR-46619)

### PR DESCRIPTION
**Pull request by JayMurdoch**
Fix for [FHIR-46619](https://jira.hl7.org/browse/FHIR-46619)
* Add invariant to Identifier.value for minimum length of 10
* Change log entry

### CODE REVIEW

File: input/resources/au-medicarecardnumber.xml [hunk 1]
```diff
@@ -44,6 +44,14 @@
         <valueString value="32788511952" />
       </example>
       <maxLength value="11" />
+      <condition value="inv-mcn-01"/>
+        <constraint>
+        <key value="inv-mcn-01"/>
+        <severity value="error"/>
+        <human value="Medicare Card Number shall be at least 10 digits"/>
+        <expression value="$this.toString().length() >= 10"/>
```
Comment (dbojicic) [inline input/resources/au-medicarecardnumber.xml:52]: toString() is not necessary as Identifier.value is already a string.

Comment (dbojicic) [reply input/resources/au-medicarecardnumber.xml:52]: Also, think the extension http://hl7.org/fhir/StructureDefinition/minLength is supported and could be considered. Could you please reconsider available options and provide justification for the chosen approach?

File: input/resources/au-medicarecardnumber.xml [hunk 2]
```diff
@@ -44,6 +44,14 @@
         <valueString value="32788511952" />
       </example>
       <maxLength value="11" />
+      <condition value="inv-mcn-01"/>
+        <constraint>
+        <key value="inv-mcn-01"/>
+        <severity value="error"/>
+        <human value="Medicare Card Number shall be at least 10 digits"/>
```
Comment (dbojicic) [inline input/resources/au-medicarecardnumber.xml:51]: The text specifies digits, but the expression does not enforce that the content must be numeric. 
Please confirm if the numeric validation is required and either update the text or the expression to match.

**Review by dbojicic — CHANGES_REQUESTED** [review]
Please see comments.

File: input/resources/au-medicarecardnumber.xml (hunk 1 above)
Comment (JayMurdoch) [reply input/resources/au-medicarecardnumber.xml:52]: Thanks @dbojicic. Retested minLength and confirm that it is supported. Changed to use minLength extension - removes need for invariant, clearly renders in table view stating minimum length alongside max length, assures at least 10 characters as per resolution.

File: input/pagecontent/changes.md [hunk 3]
```diff
@@ -93,6 +93,7 @@ To help implementers, only the more significant changes are listed here.
     </ul>
   </li>
   <li>Deprecated <a href="StructureDefinition-au-diagnosticrequest.html">AU Base Diagnostic Request</a> (<a href="https://jira.hl7.org/browse/FHIR-46714">FHIR-46714</a>).</li>
+  <li>Added invariant inv-mcn-01 to Identifier.value to specify minimum length of 10 characters in <a href="StructureDefinition-au-medicarecardnumber.html">AU Medicare Card Number</a> (<a href="https://jira.hl7.org/browse/FHIR-46619">FHIR-46619</a>).</li>
```
[the remark this answers is not in the document]
Comment (JayMurdoch) [reply input/pagecontent/changes.md:96]: Invariant removed, change log entry amended.

File: input/resources/au-medicarecardnumber.xml (hunk 2 above)
Comment (dbojicic) [reply input/resources/au-medicarecardnumber.xml]: Thanks Jaymee, confirmed with Brett to check for the length as per the resolution.

File: input/resources/au-medicarecardnumber.xml (hunk 1 above)
Comment (dbojicic) [reply input/resources/au-medicarecardnumber.xml]: Confirmed.

File: input/pagecontent/changes.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbojicic) [reply input/pagecontent/changes.md]: Confirmed.

**Review by dbojicic — APPROVED** [review]

**Review by brettesler-ext — APPROVED** [review]

**Pull request merged**